solutions/deploy-argocd-apps/README.md

```diff
@@ -0,0 +1,173 @@
+    wget -qO- https://github.com/open-cluster-management-io/clusteradm/releases/latest/download/clusteradm_linux_amd64.tar.gz | sudo tar -xvz -C /usr/local/bin/
+
+    sudo chmod +x /usr/local/bin/clusteradm
+    ```
+
+## Setup the clusters
+
+1. Edit the kind configuration files, including `cluster1-config.yaml` and `cluster2-config.yaml`, for managed clusters to update the value in `apiServerAddress` to match your private IP address, which makes the kube apiserver of the managed clusters accessable for Argo CD running on the hub cluster. 
+
+2. Run `./setup-ocm.sh`, you will see two clusters registered on the hub cluster.
+
+    ```
+    NAME       HUB ACCEPTED   MANAGED CLUSTER URLS                  JOINED   AVAILABLE   AGE
+    cluster1   true           https://cluster1-control-plane:6443                        18s
+    cluster2   true           https://cluster2-control-plane:6443                        1s
+    ```
+
+## Install Argo CD
+
+1. Install Argo CD on the OCM hub cluster
+
+    ```bash
+    kubectl create namespace argocd
+    kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
+    ```
+
+    See more informatin from [Argo CD Getting Started](https://argo-cd.readthedocs.io/en/stable/getting_started/).
+
+2. Confirm that all pods are running.
+
+    ```bash
+    $ kubectl -n argocd get pods
+    argocd-application-controller-0                    1/1     Running        0          22s
+    argocd-applicationset-controller-7f466f7cc-9hdlq   1/1     Running        0          22s
+    argocd-dex-server-54cd4596c4-t5sb5                 1/1     Running        0          22s
+    argocd-notifications-controller-8445d56d96-8745z   1/1     Running        0          22s
+    argocd-redis-65596bf87-lkrg2                       1/1     Running        0          22s
+    argocd-repo-server-5ccf4bd568-hxlzw                1/1     Running        0          22s
+    argocd-server-7dff66c8f8-qxzp9                     1/1     Running        0          22s
+    ```
+
+3. Download Argo CD CLI
+
+    Download the latest version of Argo CD CLI `argocd` from https://github.com/argoproj/argo-cd/releases/latest. More detailed installation instructions can be found via the [CLI installation documentation](https://argo-cd.readthedocs.io/en/stable/cli_installation/).
+
+## Integrate OCM with Argo CD
+
+1. Register the managed clusters to Argo CD
+
+    Start port forwarding to expose Argo CD API server on the hub cluster.
+
+    ```bash
+    kubectl port-forward svc/argocd-server -n argocd 8080:443
+    ```
+
+    Login Argo CD using the CLI
+
+    ```bash
+    export ADMIN_PASS=$(kubectl -n argocd get secret argocd-initial-admin-secret -o=jsonpath='{.data.password}' | base64 -d)
+    argocd login localhost:8080 --username=admin --password="${ADMIN_PASS}" --insecure
+    ```
+
+    Register the two OCM managed clusters to Argo CD
+
+    ```bash
+    argocd cluster add kind-cluster1 --name=cluster1
```

Comment:
will we be able to use a script to do this? or use managed-sa?

@mikeshng @xiangjingli when we have the ability to run argo with ManifestWork, this steps will be eased, right?

Reply:
Done. A separated script , `setup-argocd.sh`, is create to integrate Argo CD with OCM.

Reply:
@qiujian16   The solution is actually the push model of the argocd integration. Basically there is only one argocd server running on the hub cluster.  The argocd api server is exposed to localhost , then the argocd CLI is able to log into the argocd server, then uses argocd CLI to import the managed cluster secrets to the argocd api server NS, then create a applicationSet to directly deploy resources to the  remote managed clusters. 

In the new pull model of argocd integration we are working on, each managed cluster will run a local argocd  server. user needs to create a argocd application template on hub, and the argocd application template will be delivered to each managed cluster via manifestwork. Then the argocd server running on each managed cluster is triggered to deploy resources locally.  There is no need to import all managed cluster secrets to the ArgoCD api server. and we don't reply on argocd cli either.

With that being said, the push model and pull model don't conflict each other.  I think we can add another new solution to show how to deploy argocd app with pull model once it is available in upstream.